src/browser.ts

```diff
@@ -126,17 +126,27 @@ export class GrpcClient {
           return;
         }
       }
-      const servicePort = opts.port || 443;
-      const serviceNameArray: string[] = [];
-      let currServicePart = method.parent;
-      while (currServicePart.name !== '') {
-        serviceNameArray.unshift(currServicePart.name);
-        currServicePart = currServicePart.parent;
+      let servicePort = opts.port;
+      if (!servicePort) {
+        if (service.options && service.options['(google.api.default_host)']) {
+          servicePort = service.options['google.api.default_host'].replace(
```

Comment:
What if `google.api.default_host` does not have a port? I think the logic should be

```js
const match = service.options['google.api.default_host'].match(/:(\d+)$/);
if (match) ...
```

Here, the `:(\d+)$` tells it to see if the string ends with a colon followed by one or more digits (which look like a port). Since they are in brackets, they will be extracted and can be taken from the match:

```
> s = "hostname:123"
'hostname:123'
> match = s.match(/:(\d+)$/)
[ ':123', '123', index: 8, input: 'hostname:123', groups: undefined ]
> match[1]
'123'
```

Reply:
After some thinking... we'll need both the hostname and the port. So the regex might be `/^(.*):(\d+)$/` and then the hostname would be `match[1]` and the port would be `match[2]`.